hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/ContentGenerator.java

```diff
@@ -70,16 +94,28 @@ public void write(OutputStream outputStream) throws IOException {
       if (copyBufferSize == 1) {
         for (int i = 0; i < curSize; i++) {
           outputStream.write(buffer[i]);
+          flushOrSync(outputStream);
         }
       } else {
         for (int i = 0; i < curSize; i += copyBufferSize) {
           outputStream.write(buffer, i,
               Math.min(copyBufferSize, curSize - i));
+          flushOrSync(outputStream);
         }
       }
     }
   }
 
+  private void flushOrSync(OutputStream outputStream) throws IOException {
+    if (outputStream instanceof FSDataOutputStream) {
+      if (hSync) {
+        ((FSDataOutputStream) outputStream).hsync();
+      } else if (hFlush) {
+        ((FSDataOutputStream) outputStream).hflush();
+      }
+    }
```

Comment:
It would be great if we could avoid this runtime check/cast by using inheritance and/or generics.

Reply:
If we try to implement a check using inheritance then there would be a lot more code that will be added, hence I believe we could leave out this one.